CONTRIBUTING.md

```diff
@@ -0,0 +1,231 @@
+automatic linter in place in our CI pipeline which runs whenever code is pushed to a branch. You'll notice a check going green or red at the very bottom when viewing your open PR. The check is called "Lint with flake8". If after a `git push` you notice the check gone red, click on it and you'll see details of what the linter detected you should correct.
+
+`flake8` can be used directly on your code as well. Install it to your dev-environment:
+
+```
+pip install flake8
+```
+
+Run with:
+
+`flake8 the_file_being_worked_on.py`
+
+or leave out the filename to run it on all files and subdirectories.
+
+Some more style and coding requirements:
+
+- Pythonism: `synadm` is a Python project and as such tries to use some of the very handy features of the language. Some  of those might not be familiar to you since often we realize that contributors are professional programmers used to other languages or even are admins who are rather at home on the shell than in Python code. That is certainly all fine and we are trying to support you with refactoring your code to become more pythonic.
+- That said, we would still like `synadm` to stay approachable to Python beginners and would encourage contributors to try not to overcomplicate things.
+- We require a line length of 79 characters in Python code.
+  - Some editors have handy commands to quickly reformat multi-line text (e.g help texts of options). For example in `vim`, lines can be marked using `shift+V` and then reformatted by typing `gq`
+  - For code passages, staying within this boundary and at the same time keeping code well readable, this can be achieved using PEP8's many [line continuation suggestions](https://peps.python.org/pep-0008/#code-lay-out).
+- We don't require a maximum line length in Markdown files, for example _this document_ or `README.md`. Please omit line-breaks in sentences, paragraphs, and so on, when editing these docs.
+- If you'd like to tidy up your commits or reduce commit count we welcome the use of `git commit --amend` and `git rebase`. If you'd like us to test your code or work together within your feature branch, be aware of that force pushing might complicate things and should rather be omitted in that case.
+
+
+## Maintainers Notes
+
+This section is a checklist for maintainers of the `synadm` project. Still there might be helpful information applying to contributors as well, thus we made it public.
+
+- We keep a branch named `dev` consistently existing that may be used for random things. If this branch is in use because a PR was opened from it, it's occupied and can't be used by others.
+- Speaking of branches, we are not required to create branches in our own forks, we can just create them directly within the `synadm` repo. Checking out branches for e.g testing a feature in development is much easier than cloning and checking out from another maintainers fork.
```

Comment:
```suggestion
- Collaborators are not required to create branches in their own fork. Checking out branches already on the [JOJ0/synadm](https://github.com/JOJ0/synadm) repository is much easier than cloning and checking out from another maintainers fork.
```

It's definitely possible to checkout another repo:
```
git remote add <remote_name> <URL>
git fetch <remote_name>
git checkout <whatever>
```

It's actually what I do for my local clone (only the second and third step, first is only necessary per repo/remote).

Reply:
Sure, its just a little less things to do when a branch is directly in the same repo. I don't see a downside of having everythin in one repo and not having to switch remotes. What do you think?

(my wording was "much easier", not "impossible" :-))

Reply:
One more thing why I think it's more convenient. After some time I tend to get several remotes from several maintainers and need to clean up regularily. At least the branches that don't have a special remote I can rely on are our own owns and I don't need to fetch any particular remote, it will be done with a simple `git fetch`, but yeah again: Wording is "Collaborators are not required..." is the wording. We can still do it when there is good reasons. One thing where it's nice to have one's own fork are github acitons automations, it helps to know you would not mess with the automation of the core repo itself.